src/entity.lua

```diff
@@ -0,0 +1,44 @@
+-- This is a base class representing any moving entity
+-- The idea is to inherit this class and overwrite any
+-- default behavior when designing a moving object for example 
+-- a player, an enemy or a moving obstacle.
+
+Entity = Object:extend()
+
+
+-- Constructor
+function Entity:new(x,y,w,h)
+    
+    -- The cordinations of the top left point of the starting position
+    -- In Love the top left side of the window is the point (0,0)
+    self.x = x   
+    self.y = y
+
+    -- Hight and width of the entity
+    self.h = h
+    self.w =w
+
+end
+
+function Entity:move(ux,uy,displacement)
+    --|| Runs the update method for WASD keys ||--
+
+    if ux ~= 0 or uy ~= 0 then
+        local magnitude = math.sqrt(ux*ux + uy*uy)
+        ux, uy = ux/magnitude, uy/magnitude  -- Normalise the unit vector
+      end
+      
+    self.x = self.x + ux*displacement
+    self.y = self.y + uy*displacement
+
+end
+
+-- Default update and draw functions
+function Entity:update()
+    love.graphics.polygon( "line", self.x, self.y+self.h, self.x+(self.w/2), self.y, self.x+self.w, self.y+self.h )
```

Comment:
I think this is wrong. 
```suggestion
    function Entity:update(dt)
        -- Placeholder --
    end
```

Reply:
It's the default drawing, of an entity without a specific update function. It makes it easier to create entities without taking graphics into account. It draws triangle pointed upwards. 
Not implementing this function would mean every entity will have to implement update and draw functions, which is desired in the end product, but not during the entity development when you need to see weather ideas work or not.

Reply:
I think every declaration should have the basic `new()`, `update()` and 'draw()` methods. Although it's a little extra work, it makes a massive difference in readability. Having a drawing function in an update method could be really bad for debugging.

I agree there there should be a default `draw()` method, but there shouldn't be any drawing done in an update method.

I can just see myself pulling my hair out going "why is there this f-ing triangle on my screen?!" at 1am on Saturday.

Reply:
Sorry, I meant for you to keep the `draw()` function in. I liked that. It was only the `update()` method that I didn't like. Oh well we can leave it for now.